ports/espressif/common-hal/sdioio/SDCard.c

```diff
@@ -0,0 +1,223 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2024 Jacob Rigby
+//
+// SPDX-License-Identifier: MIT
+
+#include <stdbool.h>
+#include "esp_err.h"
+
+#include "driver/sdmmc_host.h"
+#include "ports/espressif/esp-idf/components/sdmmc/include/sdmmc_cmd.h"
+
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/util.h"
+#include "shared-bindings/sdioio/SDCard.h"
+#include "py/mperrno.h"
+#include "py/runtime.h"
+
+#include "esp_log.h"
+static const char *TAG = "SDCard.c";
+
+static bool sdio_host_init = false;
+
+static void common_hal_sdioio_sdcard_check_for_deinit(sdioio_sdcard_obj_t *self) {
+    if (common_hal_sdioio_sdcard_deinited(self)) {
+        raise_deinited_error();
+    }
+}
+
+static int check_pins(const mcu_pin_obj_t *clock, const mcu_pin_obj_t *command, const uint8_t num_data, const mcu_pin_obj_t **data) {
+    if (CONFIG_SOC_SDMMC_USE_GPIO_MATRIX) {
+        // ESP32-S3 and P4 can use any pin for any SDMMC func in either slot
+        // Default to SLOT_1 for SD cards
+        ESP_LOGI(TAG, "Using chip with CONFIG_SOC_SDMMC_USE_GPIO_MATRIX");
+        return SDMMC_HOST_SLOT_1;
```

Comment:
Do you want to track both slots so you can use two cards?

Reply:
Makes sense, I'll start working on an update...

Reply:
Looking at the code, it looks to me like the slot is determined by the  command and clock pins. there also appears to be data pin requirements depending on which slot is being used. I haven't dug through the Espressif documents to verify but if you think that's more restrictive than the peripheral requires (which the comment does suggest) I can try and figure that out.

I believe, currently both slots could be used by selecting the associated gpio pins for each slot?

Reply:
I've added the code to track the slot usage but ended up defaulting to slot 1 because using slot 0 isn't working. I've spent a lot of time staring at the code and tweaking little things but so far I'm stumped. The code still seems to mount an SD card fine using slot 1.